libraries/standard/json/utest/json_utest.c

```diff
@@ -0,0 +1,1153 @@
+    TEST_ASSERT_EQUAL_STRING_LEN( QUERY_ANSWER, outValue, QUERY_ANSWER_LENGTH );
+
+    jsonStatus = JSON_Search( JSON_DOC_MULTIPLE_VALID_ESCAPES,
+                              JSON_DOC_MULTIPLE_VALID_ESCAPES_LENGTH,
+                              COMPLETE_QUERY_KEY,
+                              COMPLETE_QUERY_KEY_LENGTH,
+                              JSON_QUERY_SEPARATOR[ 0 ],
+                              &outValue,
+                              &outValueLength );
+    TEST_ASSERT_EQUAL( JSONSuccess, jsonStatus );
+    TEST_ASSERT_EQUAL( MULTIPLE_VALID_ESCAPES_LENGTH, outValueLength );
+    TEST_ASSERT_EQUAL_STRING_LEN( MULTIPLE_VALID_ESCAPES,
+                                  outValue,
+                                  MULTIPLE_VALID_ESCAPES_LENGTH );
+
+    jsonStatus = JSON_Search( JSON_DOC_LEGAL_UTF8_BYTE_SEQUENCES,
+                              JSON_DOC_LEGAL_UTF8_BYTE_SEQUENCES_LENGTH,
+                              COMPLETE_QUERY_KEY,
+                              COMPLETE_QUERY_KEY_LENGTH,
+                              JSON_QUERY_SEPARATOR[ 0 ],
+                              &outValue,
+                              &outValueLength );
+    TEST_ASSERT_EQUAL( JSONSuccess, jsonStatus );
+    TEST_ASSERT_EQUAL( LEGAL_UTF8_BYTE_SEQUENCES_LENGTH, outValueLength );
+    TEST_ASSERT_EQUAL_STRING_LEN( LEGAL_UTF8_BYTE_SEQUENCES,
+                                  outValue,
+                                  LEGAL_UTF8_BYTE_SEQUENCES_LENGTH );
+
+    jsonStatus = JSON_Search( JSON_DOC_LEGAL_UNICODE_ESCAPE_SURROGATES,
+                              JSON_DOC_LEGAL_UNICODE_ESCAPE_SURROGATES_LENGTH,
+                              COMPLETE_QUERY_KEY,
+                              COMPLETE_QUERY_KEY_LENGTH,
+                              JSON_QUERY_SEPARATOR[ 0 ],
+                              &outValue,
+                              &outValueLength );
+    TEST_ASSERT_EQUAL( JSONSuccess, jsonStatus );
+    TEST_ASSERT_EQUAL( LEGAL_UNICODE_ESCAPE_SURROGATES_LENGTH, outValueLength );
+    TEST_ASSERT_EQUAL_STRING_LEN( LEGAL_UNICODE_ESCAPE_SURROGATES,
+                                  outValue,
+                                  LEGAL_UNICODE_ESCAPE_SURROGATES_LENGTH );
+
+    jsonStatus = JSON_Search( JSON_DOC_UNICODE_ESCAPE_SEQUENCES_BMP,
+                              JSON_DOC_UNICODE_ESCAPE_SEQUENCES_BMP_LENGTH,
+                              COMPLETE_QUERY_KEY,
+                              COMPLETE_QUERY_KEY_LENGTH,
+                              JSON_QUERY_SEPARATOR[ 0 ],
+                              &outValue,
+                              &outValueLength );
+    TEST_ASSERT_EQUAL( JSONSuccess, jsonStatus );
+    TEST_ASSERT_EQUAL( UNICODE_ESCAPE_SEQUENCES_BMP_LENGTH, outValueLength );
+    TEST_ASSERT_EQUAL_STRING_LEN( UNICODE_ESCAPE_SEQUENCES_BMP,
+                                  outValue,
+                                  UNICODE_ESCAPE_SEQUENCES_BMP_LENGTH );
```

Comment:
How about test cases in which  if only FIRST_QUERY_KEY is provided as the key?

Reply:
It shouldn't change the coverage but added an additional test case for this